client/src/lib/api.js

```diff
@@ -0,0 +1,165 @@
+export async function fetchCatalog() {
+  const res = await fetch(('/api/inventory'));
+  if (!res.ok) throw new Error(`fetch Error ${res.status}`);
+  return await res.json();
+}
+
+export async function fetchCard(cardId) {
+  const res = await fetch(`/api/inventory/${cardId}`);
+  if (!res.ok) throw new Error(`fetch Error ${res.status}`);
+  return await res.json();
+}
+
+export async function signUpOrIn(action, username, password, firstName, lastName, email) {
+  let body = { username, password, firstName, lastName, email };
+  if (action === 'sign-in') {
+    body = { username, password };
+  }
```

Comment:
Because I'm weirdly obsessed with always using `const` variables, I would have written this as:
```
const body = { username, password, ...(action === 'sign-up' ? {firstName, lastName, email} : {})};
```